### ISSUE TITLE
macOS vm_stat call breaks Linux CI + hardcoded vocab size in polyhash training

### ISSUE THREAD

**Issue by MatoTeziTanka** [asuramaya/chronohorn#12]
## Summary

Two platform/correctness issues found in the latest code:

### 1. `dispatch.py:280-288` — macOS-only `vm_stat` breaks Linux CI

```python
def parse_vm_stat_bytes() -> dict[str, Any]:
    output = capture_checked(["vm_stat"])
```

`vm_stat` is a macOS-only command. This function is called by `build_fleet_state()` (line 488) when `need_local=True`. On Linux CI (Ubuntu), this crashes — confirmed by Codex running the test suite on the CI runner.

Fix: guard with platform check:
```python
if platform.system() != "Darwin":
    return {"error": "vm_stat only available on macOS"}
```

### 2. `train_polyhash.py:613,664,711` — hardcoded `1024` vocab size

Three locations reshape logits to `(-1, 1024)` instead of using the model's configured vocab size:

```python
loss = F.cross_entropy(logits[:, :-1].reshape(-1, 1024), b[:, 1:].reshape(-1))
```

The config supports `vocab_size` as a configurable field, but the training script hardcodes it. This will silently compute wrong loss if `vocab_size != 1024`.

Fix: use `logits.size(-1)` or `config.vocab_size`:
```python
loss = F.cross_entropy(logits[:, :-1].reshape(-1, logits.size(-1)), b[:, 1:].reshape(-1))
```
